spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -398,12 +399,14 @@ static List<String> findOrphanFiles(
     spark.sparkContext().register(conflicts);
 
     Column joinCond = actualFileIdentDS.col("path").equalTo(validFileIdentDS.col("path"));
-
-    List<String> orphanFiles =
-        actualFileIdentDS
-            .joinWith(validFileIdentDS, joinCond, "leftouter")
-            .mapPartitions(new FindOrphanFiles(prefixMismatchMode, conflicts), Encoders.STRING())
-            .collectAsList();
+    PatriciaTrie<String> files = new PatriciaTrie<>();
+    actualFileIdentDS
+        .joinWith(validFileIdentDS, joinCond, "leftouter")
+        .mapPartitions(new FindOrphanFiles(prefixMismatchMode, conflicts), Encoders.STRING())
+        .foreach(
+            path -> {
+              files.put(path, null);
```

Comment:
Foreach on this local reference won't work, in the end we have to collect it at which point all the string paths are materialized in memory.

Reply:
I think there's an API for an iterator on the results, may be worth trying that and adding to the structure